(AUR) failed to compile

```
==> Lancement de check()...
scons: Reading SConscript files ...
building release: v0.4.r205.gcdea3bb
debug flag enabled: False
Checking for pkg-config... (cached) yes
Checking for gtkmm-3.0 >= 3.10... (cached) yes
Checking for glibmm-2.4... (cached) yes
Checking for gmime-2.6 >= 2.6.18... (cached) yes
Checking for webkitgtk-3.0... (cached) yes
Checking for libsass... (cached) yes
Checking for C library notmuch... (cached) yes
Checking for C function notmuch_database_get_revision()..(cached) yes
Checking for C++ library boost_filesystem... (cached) yes
Checking for C++ library boost_system... (cached) yes
Checking for C++ library boost_program_options... (cached) yes
writing src/build_config.hh..
scons: done reading SConscript files.
scons: Building targets ...
Running tests in test/test_theme
Running 1 test case...
[info ] (00:01:35): welcome to astroid! - v0.4.r205.gcdea3bb
[debug] (00:01:35): utf8: 1, UTF-8
[info ] (00:01:35): cf: loading test config.
[TEST ] (00:01:35): cf: using home directory: /tmp/yaourt-tmp-hugo/aur-astroid/src/astroid/test/test_home
[info ] (00:01:35): cf: test config, loading defaults.
[info ] (00:01:35): date: init.
[info ] (00:01:35): ac: initializing accounts..
[info ] (00:01:35): ac: setup account: charlie for Charlie Root (default: 1)
[info ] (00:01:35): global actions: set up.
[info ] (00:01:35): poll: setting up.
[debug] (00:01:35): poll: interval: 0
[info ] (00:01:35): poll: periodic polling disabled.
[debug] (00:01:35): theme: loading..
[debug] (00:01:35): tv: testing version: 2
[debug] (00:01:35): tv: testing version: 2
[info ] (00:01:35): theme: processing: ui/thread-view.scss
unknown location(0): fatal error: in "Theme/loading_theme": memory access violation at address: 0x023ec970: no mapping at fault address
test/test_theme.cc(16): last checkpoint: "loading_theme" entry.

*** 1 failure is detected in the test module "TestTheme"
scons: *** [test/test_theme.passed] Error 1
scons: building terminated because of errors.
```

Would like to reproduce, but I even don't get that far.
It already hangs at compiling the first test.
```
In file included from test/test_generic.cc:5:
test/test_common.hh:9:3: error: 'Astroid' is not a class, namespace, or enumeration
  Astroid::astroid = new Astroid::Astroid ();
  ^
test/test_common.hh:9:3: error: reference to 'Astroid' is ambiguous
  Astroid::astroid = new Astroid::Astroid ();
  ^
src/log.hh:21:11: note: candidate found by name lookup is 'Astroid'
namespace Astroid {
          ^
src/astroid.hh:14:9: note: candidate found by name lookup is 'Astroid::Astroid'
  class Astroid {
        ^
In file included from test/test_generic.cc:5:
test/test_common.hh:9:26: error: 'Astroid' is not a class, namespace, or enumeration
  Astroid::astroid = new Astroid::Astroid ();
                         ^
test/test_common.hh:9:26: error: reference to 'Astroid' is ambiguous
  Astroid::astroid = new Astroid::Astroid ();
                         ^
src/log.hh:21:11: note: candidate found by name lookup is 'Astroid'
namespace Astroid {
          ^
src/astroid.hh:14:9: note: candidate found by name lookup is 'Astroid::Astroid'
  class Astroid {
        ^
In file included from test/test_generic.cc:5:
test/test_common.hh:14:10: error: 'Astroid' is not a class, namespace, or enumeration
  delete Astroid::astroid;
         ^
test/test_common.hh:14:10: error: reference to 'Astroid' is ambiguous
  delete Astroid::astroid;
         ^
src/log.hh:21:11: note: candidate found by name lookup is 'Astroid'
namespace Astroid {
          ^
src/astroid.hh:14:9: note: candidate found by name lookup is 'Astroid::Astroid'
  class Astroid {
        ^
6 errors generated.
```

I can fix that by removing "using namespace Astroid" from
test/test_common.h. For the other tests, I then need to add "using
namespace Astroid", again.
But then I end up here:

```
test/test_non_existant_file.cc:57:9: error: reference to 'log' is ambiguous
        log << test << "trying to open deleted file." << endl;
```
NOTE: "using namespace" in a header file is really bad ;)
using namespace std; even more.

@gauteh would you mind if I went along and cleaned that up? I think I
also saw some unneeded includes which I could try to get then.

Excerpts from Hugo Roy's message of Januar 16, 2016 12:02 :
> ==> Lancement de check()...
> scons: Reading SConscript files ...
> building release: v0.4.r205.gcdea3bb
> debug flag enabled: False
> Checking for pkg-config... (cached) yes
> Checking for gtkmm-3.0 >= 3.10... (cached) yes
> Checking for glibmm-2.4... (cached) yes
> Checking for gmime-2.6 >= 2.6.18... (cached) yes
> Checking for webkitgtk-3.0... (cached) yes
> Checking for libsass... (cached) yes
> Checking for C library notmuch... (cached) yes
> Checking for C function notmuch_database_get_revision()..(cached) yes
> Checking for C++ library boost_filesystem... (cached) yes
> Checking for C++ library boost_system... (cached) yes
> Checking for C++ library boost_program_options... (cached) yes
> writing src/build_config.hh..
> scons: done reading SConscript files.
> scons: Building targets ...
> Running tests in test/test_theme
> Running 1 test case...
> [info ] (00:01:35): welcome to astroid! - v0.4.r205.gcdea3bb
> [debug] (00:01:35): utf8: 1, UTF-8
> [info ] (00:01:35): cf: loading test config.
> [TEST ] (00:01:35): cf: using home directory: /tmp/yaourt-tmp-hugo/aur-astroid/src/astroid/test/test_home
> [info ] (00:01:35): cf: test config, loading defaults.
> [info ] (00:01:35): date: init.
> [info ] (00:01:35): ac: initializing accounts..
> [info ] (00:01:35): ac: setup account: charlie for Charlie Root (default: 1)
> [info ] (00:01:35): global actions: set up.
> [info ] (00:01:35): poll: setting up.
> [debug] (00:01:35): poll: interval: 0
> 
> ```
> [info ] (00:01:35): poll: periodic polling disabled.
> [debug] (00:01:35): theme: loading..
> [debug] (00:01:35): tv: testing version: 2
> [debug] (00:01:35): tv: testing version: 2
> [info ] (00:01:35): theme: processing: ui/thread-view.scss
> unknown location(0): fatal error: in "Theme/loading_theme": memory access violation at address: 0x023ec970: no mapping at fault address
> test/test_theme.cc(16): last checkpoint: "loading_theme" entry.
> 
> *** 1 failure is detected in the test module "TestTheme"
> scons: *** [test/test_theme.passed] Error 1
> scons: building terminated because of errors.
> ```
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/gauteh/astroid/issues/60

--- a/src/modes/thread_view/theme.cc
+++ b/src/modes/thread_view/theme.cc
@@ -132,9 +132,10 @@ namespace Astroid {
     }
 
     const char * output = sass_context_get_output_string(context);
+    ustring output_str(output);
     sass_delete_file_context (file_ctx);
 
-    return ustring (output);
+    return output_str;
   }
 
   bool Theme::check_theme_version (bfs::path p) {

